components/supervisor/pkg/supervisor/services.go

```diff
@@ -38,11 +38,42 @@ type RegisterableRESTService interface {
 	RegisterREST(mux *runtime.ServeMux, grpcEndpoint string) error
 }
 
+type ideReadyService struct {
```

Comment:
nit: I'd call this `ideReadyState` instead of service. So far in supervisor services are things that can be consumed by someone from the outside (e.g. status service).